targets/unity-v2/Testing/Tests/PersistentSockets/PubSubTest.cs

```diff
@@ -0,0 +1,103 @@
+#if ENABLE_PLAYFABPUBSUB_API
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayFab.UUnit
+{
+    public class PubSubTest : UUnitTestCase
+    {
+        static PlayFab.EventsModels.EntityKey _MyEntityKey = null;
+        static string _previousTitleId = null;
+
+        private PubSub pubSub;
+        private Entity entity;
+
+        private int testTicks = 0;
+        private int playstreamTickDelay = 150;
```

Comment:
Ticks is a fairly unreliable time measurement.
If I were writing this, I would typically define readonly Time elements, specifically TimeSpan: writeDelay, and DateTime nextWrite.
That's effectively what you have with your tick vars, but if the framerate were VERY LOW for some reason (like we're using machines on aws with really low hardware resources - which is very relevant), then you might fail because your tickrate is too slow.